src/bootstrap/flags.rs

```diff
@@ -216,7 +216,27 @@ To learn more about a subcommand, run `./x.py <subcommand> -h`",
         };
 
         // Some subcommands get extra options
-        match subcommand.as_str() {
+        let run_stage = |opts: &mut Options| {
+            opts.optopt(
+                    "",
+                    "run-stage",
+                    "stage to run, in terms of which libraries it runs with (e.g. `--run-stage 1` means use `build/stage1/bin`). \
+                        This is an alias for `--stage`.",
+                    "N",
+                );
+            Some("run-stage")
+        };
+        let link_stage = |opts: &mut Options| {
+            opts.optopt(
+                    "",
+                    "link-stage",
+                    "stage to use, in terms of what libraries it is linked to (e.g. `--link-stage 1` means use `build/stage1-<component>`). \
```

Comment:
This also feels confusing, because we never link to build/stage1-component, we always link to stageN/lib/rustlib/.../lib/... which is uplifted from stageN-components. (and in stage0 that directory is called stage0-sysroot, iirc, but I might be misremembering).

Reply:
Ok, how does this look then?
```suggestion
                    "stage to use, in terms of what libraries it is linked to (e.g. `--link-stage 1` means to build `build/stage1-<component>` and uplift it to `build/stage2/lib/rustlib`). \
```